Adding Solo.io to the Partners

PetrMc opened the pull request:
*Description of changes:*

This pull request integrates Solo.io into the AWS EKS Anywhere Partners testing framework. It includes the following artifacts:

- FluxCD configuration
- Testing Job

These additions ensure that Solo.io's components are validated according to the AWS EKS Anywhere Partners' testing protocols.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

PetrMc commented:
@elamaran11  @shapirov103 - would appreciate your review and hopefully approval! Thanks!

elamaran11 commented on Validated_Partners/baremetal_Partners.txt:
```diff
@@ -15,5 +15,6 @@ suse             neuvector
 newrelic         newrelic-bundle
 perfectscale     perfectscale
 pulumi           pulumi-kubernetes-operator
+solo.io          solo-istiod
```
Remove this change, this will be done via our automation

elamaran11 commented on Validated_Partners/nutanix_Partners.txt:
```diff
@@ -1,5 +1,6 @@
  dynatrace
  kubecost
+ solo
```
Remove this change, this will be done via our automation

elamaran11 commented on Validated_Partners/snow_Partners.txt:
```diff
@@ -11,6 +11,7 @@ accuknox         kubearmor
 kubecost         cost-analyzer
 nirmata          enterprise-kyverno
 lacework         polygraph
+solo.io          solo-istiod
```
Remove this change, this will be done via our automation

elamaran11 commented on Validated_Partners/vsphere_Partners.txt:
```diff
@@ -15,5 +15,6 @@ suse             neuvector
 newrelic         newrelic-bundle
 perfectscale     perfectscale
 pulumi           pulumi-kubernetes-operator
+solo.io          solo-istiod
```
Remove this change, this will be done via our automation

elamaran11 commented on Validated_Partners/eks-local-clusters-outposts_Partners.txt:
```diff
@@ -14,5 +14,6 @@ lacework         polygraph
 suse             neuvector
 perfectscale     perfectscale
 pulumi           pulumi-kubernetes-operator
+solo.io          solo-istiod
```
Remove this change, this will be done via our automation

elamaran11 commented on eks-anywhere-common/Testers/Solo.io/solo-istiod-testJob.yaml:
```diff
@@ -0,0 +1,66 @@
+                - -c
+                - >-
+                  istiod_pod_state=$(kubectl get pods -n istio-system -l app=istiod -o json | jq -r .items[].status.containerStatuses[].ready);
+                  echo Getting current istiod state.;                  
```
Please check our [Functional requirements](https://aws-samples.github.io/eks-anywhere-addons/#functional-job-requirements) for test job. Container Status does not align to our requirements of functional job.

PetrMc replied:
Thanks @elamaran11 - quick question if the test is replaced with curl call similar to SUSE [here](https://github.com/aws-samples/eks-anywhere-addons/blob/main/eks-anywhere-common/Testers/SUSE/neuvector-testJob.yaml) will it satisfy?

PetrMc replied:
@elamaran11 - please let us know if modified testJob is acceptable. Thank you!

elamaran11 replied:
@PetrMc Unfortunately this is not a good example. It was early days before we created those requirements. check other partners like Kong, NewRelic, Pulumi, Nirmata with examples which satisfies all requirements. Please go over the functional job reqs you will get an idea.

PetrMc replied:
@elamaran11 - please let me know if the new re-written functional test will satisfy. Thank you and apologies for multiple commits!

elamaran11 reviewed (changes requested):
@PetrMc Thankyou for the PR. Have some comments on the PR.

elamaran11 reviewed (changes requested):
@PetrMc Please read requirement number 3. Please rework on the functional test job pls.
```
Healthchecks, service endpoints checks or any other technical checks do not represent sufficient coverage required for the functional test
```

elamaran11 reviewed (commented):
@PetrMc Looks good at the moment. Our team will validate and get back in our EKS-A environments.

elamaran11 commented:
@mikemcd3912 Please validate in our lab.

mikemcd3912 commented:
Hi @PetrMc,  

After validating in our baremetal lab I noted that the testjob script is set up to only run once, so we would like to see that updated to a Cronjob so that it can test at least daily. See [Komodor ](https://github.com/aws-samples/eks-anywhere-addons/blob/main/eks-anywhere-common/Testers/Komodor/komodor-testjob.yaml) or [Kong](https://github.com/aws-samples/eks-anywhere-addons/blob/main/eks-anywhere-common/Testers/Kong/test-job.yaml) for examples of this  

In addition to the recurring job runs, we encountered an issue with the cleanup stage on that pod. solo-istiod-healthtest pod log output is as follows:

 $ kubectl logs -n istio-system solo-istiod-healthtest-c46qx                                                                                            ─╯
Deploying custom Istio Ingress Gateway...
deployment.apps/custom-ingressgateway created
NAME                    READY   UP-TO-DATE   AVAILABLE   AGE
custom-ingressgateway   0/1     1            0           0s
Waiting for the custom ingress gateway deployment to be ready...
Waiting for deployment "custom-ingressgateway" rollout to finish: 0 of 1 updated replicas are available...
deployment "custom-ingressgateway" successfully rolled out
Deploying test gateway...
gateway.networking.istio.io/httpbin-gateway created
NAME              AGE
httpbin-gateway   0s
Deploying test VirtualService...
virtualservice.networking.istio.io/httpbin-virtualservice created
NAME                     GATEWAYS              HOSTS                     AGE
httpbin-virtualservice   ["httpbin-gateway"]   ["httpbin.example.com"]   0s
Checking Istio Ingress Gateway configuration...
        "name": "httpbin.example.com:80",
         "httpbin.example.com"
Istio Ingress Gateway has received correct configuration from Istiod.
Cleaning up test resources...
virtualservice.networking.istio.io "httpbin-virtualservice" deleted
gateway.networking.istio.io "httpbin-gateway" deleted
Istio functional test completed successfully.
Cleaning up resources...
Error from server (NotFound): gateways.networking.istio.io "httpbin-gateway" not found
deployment.apps "custom-ingressgateway" deleted
Cleanup completed.

PetrMc commented:
Thanks @mikemcd3912 

- Good catch on cleaning up script - it's now adjusted - thanks.
- Job has been converted to CronJob (per @elamaran11 directions we did follow [the requirements](https://aws-samples.github.io/eks-anywhere-addons/#functional-job-requirements) that are specifically state in #5 "Functional test must be implemented as a **Kubernetes Job**" - you might want to adjust the docs.

Hopefully the new commit addresses all the concerns. Thank you!

mikemcd3912 reviewed (commented):
@PetrMc Thank you for the quick adjustment, and I appreciate the feedback on the docs! I can see how that current wording would be a bit unclear so we'll work to update that going forward

@elamaran11 
- vmware testing successful, no issues noted with addon deployment or test job after the cleanup script adjustment
- baremetal testing reperformed successfully, no issues noted with addon deployment or test job after the cleanup script adjustment
- local testing successful, no issues noted with addon deployment or test job after the cleanup script adjustment

elamaran11 reviewed (approved):
@PetrMc Thankyou for working with us on the Conformitron program to validate `solo.io`. 
@mikemcd3912 Thankyou for validating in our labs. LGTM to merge.
